openvino/cc/scripts/ccheader.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+import argparse, csv
+from pathlib import Path
+
+Domain = ['CC0_',
+          'CC1_',
+          'CC2_']
+
+FILE_HEADER = "#pragma once\n\n"
+
+FILE_FOOTER = "\n"
+
+ENABLED_SCOPE_FMT = "#define %s_%s 1\n"
+ENABLED_SWITCH_FMT = "#define %s_%s 1\n#define %s_%s_cases %s\n"
+ENABLED_FACTORY_INSTANCE_FMT = "#define %s_%s 1\n"
+
+class Scope:
```

Comment:
Why don't use abstract base class for all this classes?

Reply:
refactored